main/src/library/Http.flix

```diff
@@ -0,0 +1,48 @@
+/*
+ *  Copyright 2024 Holger Dal Mogensen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// The effect used to interact with the HTTP protocol.
+///
+eff Http {
+    pub def get(url: String, headers: Map[String, String]): Http.Response
+}
+
+mod Http {
+
+    ///
+    /// A response from an HTTP request.
+    ///
+    /// The response consists of: `(status, statusText, headers, body)`
+    ///
+    pub enum Response(Int32, String, String, String)
```

Comment:
Can the response be binary? Or would it then be encoded somehow?

Reply:
The whole message is always encoded as a string so that would be the most general case.

However, it looks like there are different cases: https://developer.mozilla.org/en-US/docs/Web/HTTP/Messages#body_2

There's 
- the regular single string
- the stream of unknown length, (but this is disallowed with HTTP/2)
- the multiple-resource body, with multiple sections delimited by `--myBoundary`, but even MDN admits that these are rare

I definitely think we can just handle the third case by letting users manually decode the string, but I'm not sure about the second case.

Reply:
OK. Lets just support a string for now.